test/redis/connection.js

```diff
@@ -195,6 +162,114 @@ describe('redis/connection:', () => {
     });
   });
 
+  redisConnection.methods.forEach(method => {
+    if (method === 'update') {
+      return;
+    }
+
+    const clientMethod = `${method}Async`;
+
+    describe(`redisConnection.${method}:`, () => {
+      let result;
+
+      beforeEach(async () => {
+        redisClient[clientMethod] = sinon.spy(() => Promise.resolve(`mock ${method} result`));
```

Comment:
Should we be ensuring these methods exist on the real client so that we catch API expectation mismatches early? I suppose the integration tests take care of that.

Reply:
> I suppose the integration tests take care of that.

Precisely.